src/parse/parse.c

```diff
@@ -6,64 +6,147 @@
+        enter_file(fd);
+        printf("%d\n",fd);
+        //printf("\n");
+        return SUCCESS;
+    }
+    return FAILURE;
 }
-/*
-int parse(t_data *data)
-{
-    int status;
-    status = 0;
 
-    //status = tokens_parsing(data->token);
-    return (status);
+
+int handle_input_redirection(t_token *token, t_token *next_token)
+{
+    if (ft_strcmp(token->value, "<") == 0)
+    {
+        open(next_token->value, O_RDONLY);
+        printf("\n"); // idk if we actually need to print this new line
+        return (0);
+    }
+    return FAILURE;
 }
-*/
 
-		// if ((head->type == REDIRECT_IN && next_type == COMMAND)
-		// 	|| (head->type == REDIRECT_IN && next_type == REDIRECT_IN)
-		// 	|| (head->type == REDIRECT_IN && next_type == REDIRECT_OUT)
-		// 	|| (head->type == REDIRECT_IN && next_type == PIPE)
-		// 	|| (head->type == REDIRECT_IN && next_type == HEREDOC)
-		// 	|| (head->type == REDIRECT_IN && next_type == APPEND)
-		// 	|| (head->type == REDIRECT_IN && next_type == FLAG))
\ No newline at end of file
+
+
+int parse_token(t_token *token)
+{
+    t_token *current_token = token;
+
+    while (current_token != NULL)
+    {
+        // Handle output redirection (">" or ">>")
+        if ((current_token->type == RED_OUT || current_token->type == APPEND)
+            && current_token->next != NULL && current_token->next->value != NULL)
+        {
+            if (current_token->prev == NULL) // No previous token
+            {
+                if (current_token->next->type == ARGUMENT)
+                    handle_output_redirection(current_token, current_token->next);
+                else
+                {
+                    printf("error 106\n");
+                    return 44;
+                }
+            }
+            else // Previous token exists
+            {
+                if (current_token->next->type == ARGUMENT && current_token->prev->type == PIPE)
+                    handle_output_redirection(current_token, current_token->next);
+                else
+                {
+                    printf("error 117\n");
+                    return 44;
+                }
+            }
+        }
+        // Handle input redirection ("<" or "<<")
+        else if ((current_token->type == RED_IN || current_token->type == HEREDOC)
+            && current_token->next != NULL && current_token->next->value != NULL)
+        {
+            if (current_token->prev == NULL) // No previous token
+            {
+                if (current_token->next->type == ARGUMENT)
+                    handle_input_redirection(current_token, current_token->next);
+                else
+                {
+                    printf("error 132\n");
+                    return 44;
+                }
+            }
+            else // Previous token exists
+            {
+                if (current_token->next->type == ARGUMENT && current_token->prev->type == PIPE)
+                    handle_input_redirection(current_token, current_token->next);
```

Comment:
we might need to rethink this logic, because this parsing is happening because execution. For now, leave it be. But just so you can see it, HERE_DOC here happens in the middle of piping and it overwrites the outfile and ignores previous stuff. The overwriting is due to the fact that I used the redirection for output ">" which overwrites and does not append.

![image](https://github.com/user-attachments/assets/deedab18-5f01-4871-8377-99742b939a51)